project-koku/koku
# [COST-3572] - actually use the constructed daily archives for OCP

## Jira Ticket

[COST-3572](https://issues.redhat.com/browse/COST-3572)

## Description

Currently we are discarding the names of the daily files that are archived for OCP. This PR uses those filenames throughout the rest of the code rather than using the original filename in the payload.

## Testing

https://github.com/project-koku/koku/pull/4546
Following the above testing instructions, but in the ephemeral cluster, ingesting the data thru ingress and verifying the payloads in minio.

1. create a source:
```
port forward to sources:
$ oc port-forward service/sources-api-svc 3000:8000

POST to create the source (this source_ref matches the payload, so don't change it)
$ curl --location 'http://localhost:3000/api/sources/v3.1/bulk_create' \
--header 'x-rh-identity: eyJpZGVudGl0eSI6eyJhY2NvdW50X251bWJlciI6IjEwMDAxIiwib3JnX2lkIjoiMTIzNDU2NyIsInR5cGUiOiJVc2VyIiwidXNlciI6eyJ1c2VybmFtZSI6InVzZXJfZGV2IiwiZW1haWwiOiJ1c2VyX2RldkBmb28uY29tIiwiaXNfb3JnX2FkbWluIjp0cnVlLCJhY2Nlc3MiOnsgfX19LCJlbnRpdGxlbWVudHMiOnsiY29zdF9tYW5hZ2VtZW50Ijp7ImlzX2VudGl0bGVkIjp0cnVlfX19Cg==' \
--header 'x-rh-sources-psk: thisMustBeEphemeralOrMinikube' \
--header 'Content-Type: application/json' \
--data '{
    "sources": [{
        "name":"ocp source",
        "source_ref": "16b9a60d-0774-4102-9028-bd28d6c38ac2",
        "source_type_name": "openshift"
    }],
    "endpoints":[],
    "authentications":[],
    "applications":[{
        "application_type_name":"cost-management",
        "source_name": "ocp source"
    }]
}'
```

2. Send the attached payload:
```
port forward ingress:
$ oc port-forward service/ingress-service 20000:8000

post the payload to ingress:
$ curl -vvvv -F "file=@ea0c83df5cdf4228bef28c6f2d1d923a.gz ;type=application/vnd.redhat.hccm.tar+tgz" -H "x-rh-identity: eyJpZGVudGl0eSI6eyJhY2NvdW50X251bWJlciI6IjEwMDAxIiwib3JnX2lkIjoiMTIzNDU2NyIsInR5cGUiOiJVc2VyIiwidXNlciI6eyJ1c2VybmFtZSI6InVzZXJfZGV2IiwiZW1haWwiOiJ1c2VyX2RldkBmb28uY29tIiwiaXNfb3JnX2FkbWluIjp0cnVlLCJhY2Nlc3MiOnsgfX19LCJlbnRpdGxlbWVudHMiOnsiY29zdF9tYW5hZ2VtZW50Ijp7ImlzX2VudGl0bGVkIjp0cnVlfX19Cg==" http://localhost:20000/api/ingress/v1/upload
```

3. log into minio:
```
get creds:
$ oc get secret env-$NAMESPACE-minio -o json | jq -r '.data | map_values(@base64d)' 

port forward minio:
$ oc port-forward service/env-${NAMESPACE}-minio 9001:9000 
```
http://localhost:9001/minio/login

see CSVs:
![image](https://github.com/project-koku/koku/assets/19751919/f93ee1af-e9d5-44f2-b81b-d0e95656fdce)

see each report type (namespace_labels, node_labels, pod_usage, storage_usage) with parquet files matching the names of the csv:
![image](https://github.com/project-koku/koku/assets/19751919/29dc42d7-fd89-40ff-b4d3-5d0dcc102de4)

again, see each report type, but daily:
![image](https://github.com/project-koku/koku/assets/19751919/97894918-d0b6-4ab3-b88e-1360350dff4f)

these filenames have the structure of `{report_type}.{day}.{manifest_id}.{count}` where the count represents payloads that split the files due to their size.


## Notes
If you run the above on main, you'll see the daily CSV files, but you will not see the correct parquet files.

...
[ea0c83df5cdf4228bef28c6f2d1d923a.gz](https://github.com/project-koku/koku/files/12528231/ea0c83df5cdf4228bef28c6f2d1d923a.gz)
